src/MongoDB.Driver.Core/Core/Connections/CommandEventHelper.cs

```diff
@@ -45,6 +46,12 @@ internal class CommandEventHelper
             "copydb"
         };
 
+        private static readonly ReadOnlyDictionary<string, Func<BsonDocument, bool>> __conditionallySecuritySensitiveCommands =
+            new ReadOnlyDictionary<string, Func<BsonDocument, bool>>(new Dictionary<string, Func<BsonDocument, bool>>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "isMaster", c => c.Contains("speculativeAuthenticate") }
+            });
+
```

Comment:
I would get rid of both of these static variables and just put the logic in the code:

    private static bool ShouldRedactMessage(string commandName, BsonDocument command)
    {
        switch (commandName.ToLowerInvariant())
        {
            case "authenticate":
            case "saslStart":
            case "saslContinue":
            case "getnonce":
            case "createUser":
            case "updateUser":
            case "copydbgetnonce":
            case "copydbsaslstart":
            case "copydb":
                return true;

            case "isMaster":
                return command.Contains("speculativeAuthenticate");

            default:
                return false;
        }
    }

Reply:
Okay :)